src/app/functions/server/dashboard.test.ts

```diff
@@ -2,7 +2,1058 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { DashboardSummary, getExposureReduction } from "./dashboard";
+import { OnerepScanResultRow } from "knex/types/tables";
+import {
+  DashboardSummary,
+  getDashboardSummary,
+  getExposureReduction,
+} from "./dashboard";
+import { SubscriberBreach } from "../../../utils/subscriberBreaches";
+
+const unresolvedBreaches: SubscriberBreach[] = [
```

Comment:
I suppose you've done it like this for a reason, but I was wondering if it would be possible to re-use (/extend where needed) the mock data generation functions here, so that the only code that's listed here is the code that actually affects the test results? Right now, readers have to scan for the relevant data in between things that do not affect the test results, like names, descriptions, or dates.

Reply:
Yea I think for predicability sake, we shouldn't rely on these mock data generation functions for tests (which are black boxes and can easily be changed by someone for other purpose, ie frontend storybook looks, and end up affecting the results of tests)

I'd prioritize the predictability of these tests, which means the input should be static and does not vary from run to run

Reply:
Yeah that's sounds fine by me. Maybe the static uninteresting parts can still be extracted into a static mock data generation function?

Reply:
What I've seen with more structured unit tests is that we'd break down these bigger `types` into smaller subtypes, and just have smaller chunks of `.json` files (test objects) pieced together to enumerate the various inputs. I think we can move towards that in the future if we have more unit tests.

I did just make the test objects smaller just now to: 1) save some space 2) readability